api/runtime/v1beta2/extensionconfig_types.go

```diff
@@ -229,8 +230,8 @@ type ExtensionConfig struct {
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	// spec is the desired state of the ExtensionConfig.
-	// +optional
-	Spec ExtensionConfigSpec `json:"spec,omitempty"`
+	// +required
```

Comment:
required because spec.ClientConfig is required

Reply:
ClientConfig has MinProperties because one of URL / Service is required (otherwise they are optional)